zerver/tests/test_users.py

```diff
@@ -120,20 +120,21 @@ def test_updating_non_existent_user(self) -> None:
         self.assert_json_error(result, 'No such user')
 
     def test_admin_api(self) -> None:
-        self.login('desdemona')
+        self.login('hamlet')
```

Comment:
I don't understand the change here; why can't we do all these operations as desdemona, the realm owner user?  

That's certainly the intent of how we wanted this test to work; I don't see why we'd want to make hamlet an admin instead.

Reply:
As desdemona was changed to owner in this commit and we check `is_admin` flag for the requesting user, that is the reason why I changed it to hamlet to  correctly check is_admin flag.

Reply:
But shouldn't `.is_admin` be True for desdemona following other changes we've made?

Reply:
But the change of `is_admin as True for owners` is in the next commit. And the next commit requires an owner for tests, so I have ordered commit in this way.

Moreover in further commits, as we have removed the restriction on removing last admin, we just require one admin in this test, so we do not require hamelt as admin anyway at the end.